src/layers/layer.cpp

```diff
@@ -525,6 +529,29 @@ void Layer::setup_pointers()
                 get_child_names(),
                 ")");
   }
+
+  // Set whether this layer will run in-place
+  if (!this->can_run_inplace() || this->distconv_enabled()) {
```

Comment:
```suggestion
  if (!this->can_run_inplace()
#ifdef LBANN_HAS_DISTCONV
      || this->distconv_enabled()
#endif // LBANN_HAS_DISTCONV
      ) {
```

Reply:
While I agree with this change in the current state of the code, a better solution might be to have `layer::distconv_enabled()` always defined, just return `false` if no distconv support. I'm testing this now, and I'll PR if everything is ok in CI.